specs/consensus.md

```diff
@@ -71,16 +86,160 @@ Consensus Rules
 
 ## Leader Selection
 
+TODO
+
 ## Fork Choice
 
+TODO
+
 ## Block Validity
 
-### State Transitions
+The validity of a newly-seen block, `block`, is determined by two components, detailed in subsequent sections:
+1. [Block structure](#block-structure): whether the block header is valid, and data in a block is arranged into a valid and matching data root (i.e. syntax).
+1. [State transition](#state-transitions): whether the application of transactions in the block produces a matching and valid state root (i.e. semantics).
+
+## Block Structure
+
+Before executing [state transitions](#state-transitions), the structure of the [block](./data_structures.md#block) must be verified.
+
+The following block fields are acquired from the network and parsed (i.e. [deserialized](./data_structures.md#serialization)). If they cannot be parsed, the block is ignored but is not explicitly considered invalid by consensus rules. Further implications of ignoring a block are found in the [networking spec](./networking.md).
+1. [block.header](./data_structures.md#header)
+1. [block.availableDataHeader](./data_structures.md#availabledataheader)
+1. [block.lastCommit](./data_structures.md#commit)
+
+If the above fields are parsed successfully, the available data `block.availableData` is acquired in erasure-coded form as [a list of share rows](./networking.md#availabledata), then parsed. If it cannot be parsed, the block is ignored but not explicitly invalid, as above.
+
+### `block.header`
+
+The [block header](./data_structures.md#header) `block.header` (`header` for short) is the first thing that is downloaded from the new block, and commits to everything inside the block in some way. For previous block `prev` (if `prev` is not known, then the block is ignored), and previous block header `prev.header`, the following checks must be `true`:
```

Comment:
> The [block header](./data_structures.md#header) `block.header` (`header` for short) is the first thing that is downloaded from the new block

If it is actively downloaded or received (e.g. via some gossip routine etc) should go in the network spec. Maybe say "processed", this implies that you have it at hand.

Reply:
The plan is actually to move that entire paragraph to the networking section, since a node won't even accept a block header unless it passes those checks.